impl/src/main/java/org/ehcache/internal/store/DefaultStoreProvider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.internal.store;
+
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourcePools;
+import org.ehcache.internal.store.disk.DiskStore;
+import org.ehcache.internal.store.heap.OnHeapStore;
+import org.ehcache.internal.store.tiering.CacheStore;
+import org.ehcache.internal.store.tiering.CacheStoreServiceConfig;
+import org.ehcache.spi.ServiceProvider;
+import org.ehcache.spi.cache.Store;
+import org.ehcache.spi.service.ServiceConfiguration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Ludovic Orban
+ */
+public class DefaultStoreProvider implements Store.Provider {
+  private ServiceProvider serviceProvider;
+
+  @Override
+  public <K, V> Store<K, V> createStore(Store.Configuration<K, V> storeConfig, ServiceConfiguration<?>... serviceConfigs) {
+    ResourcePools resourcePools = storeConfig.getResourcePools();
```

Comment:
My understanding is that a null resource pool indicates that the kind of resource you're interested in is not available.

A null ResourcePools on the other hand, may be something undesired to prevent null checks at the end of the call stack.